napari/util/tests/test_io.py

```diff
@@ -66,41 +79,63 @@ def test_irregular_images(irregular_images):
     # files. We need to be able to inspect the file shape without reading
     # it in first, then we can automatically turn stacking off when shapes
     # are irregular (and create proper dask arrays)
-    images = io.magic_read(image_files, use_dask=False, stack=False)
+    images = io.magic_imread(image_files, use_dask=False, stack=False)
     assert isinstance(images, list)
     assert len(images) == 2
     assert tuple(image.shape for image in images) == ((512, 512), (303, 384))
 
 
 def test_tiff(single_tiff):
     image_files = single_tiff
-    images = io.magic_read(image_files)
+    images = io.magic_imread(image_files)
     assert isinstance(images, np.ndarray)
     assert images.shape == (2, 15, 10)
     assert images.dtype == np.uint8
 
 
 def test_many_tiffs(single_tiff):
     image_files = single_tiff * 3
-    images = io.magic_read(image_files)
+    images = io.magic_imread(image_files)
     assert isinstance(images, da.Array)
     assert images.shape == (3, 2, 15, 10)
     assert images.dtype == np.uint8
 
 
 def test_single_filename(single_tiff):
     image_files = single_tiff[0]
-    images = io.magic_read(image_files)
+    images = io.magic_imread(image_files)
     assert images.shape == (2, 15, 10)
 
 
 @pytest.mark.skipif(not zarr_available, reason='zarr not installed')
-def test_zarr(single_tiff):
-    image_files = single_tiff * 3
-    images = io.magic_read(image_files)
+def test_zarr():
+    image = np.random.random((10, 20, 20))
+    with TemporaryDirectory(suffix='.zarr') as fout:
+        z = zarr.open(fout, 'a', shape=image.shape)
+        z[:] = image
+        image_in = io.magic_imread([fout])
+        # Note: due to lazy loading, the next line needs to happen within
+        # the context manager. Alternatively, we could convert to NumPy here.
+        np.testing.assert_array_equal(image, image_in)
+
+
+@pytest.mark.skipif(not zarr_available, reason='zarr not installed')
+def test_zarr_pyramid():
+    np.random.seed(0)
```

Comment:
Why are you including the seed here?

Reply:
i've generally put one in whenever i call np.random.random, which happens in the next line, even if it shouldn't make a difference. nice to have that known reproducibility i think. i just realized i missed the seed in the test above so i will add it in there for completeness too

Reply:
No don't do that! LOL

Reply:
We do that in scikit-image because it's easy and for historical reasons. But the correct way to do this is to report the current seed (or choose a random one and then display it) so that your input is actually random. It helps to catch bugs that depend on random input.

Reply:
hmm ok - we do this in many places in the codebase right now, so we should probably go through and systematically fix if needed - remember when we did this at scipy, it was because occasionally the vispy triangulation code was failing. i'll pull the seed from these two functions and make a new commit, but followup will be needed elsewhere